js/private/js_run_binary.bzl

```diff
@@ -107,6 +109,19 @@ def js_run_binary(
 
             This makes node binaries match the expected bazel paradigm.
 
+        run_from_execroot: Runs the `tool` entry_point out of the execroot output tree instead of the runfiles of `tool`.
+
+            Runfiles of `tool` are all hoisted to `srcs` of the underlying `run_binary` so they are included as execroot
+            inputs to the action.
+
+            Running the entry point out of the execroot output tree means that there is no conflicting runfiles
+            `node_modules` in the node_modules resolution path to confuse npm packages such as next and react which
+            don't like to be resolved in multiple node_modules trees. This more closely emulates the environment that
+            tools such as Next.js see when they are run outside of Bazel.
+
+            When True, it is recommended that the js_binary tool use the `copy_data_to_bin` default of True
```

Comment:
nit: instead of "recommended" it would be nice to error if the tool doesn't copy its data to bin

Reply:
it can be done as long there is an opt-out to disable the validation

Reply:
✅